src/Microsoft.Health.Fhir.Stu3.Core/Features/Search/Expressions/Expression.cs

```diff
@@ -4,14 +4,15 @@
 // -------------------------------------------------------------------------------------------------
 
 using System.Collections.Generic;
+using EnsureThat;
 using Hl7.Fhir.Model;
 
 namespace Microsoft.Health.Fhir.Core.Features.Search.Expressions
 {
     /// <summary>
     /// Represents an expression.
     /// </summary>
-    public abstract class Expression
+    public abstract class Expression : IExpression
```

Comment:
We shouldn't need to move the expression stuff since it should be generic enough to be version agnostic.

Reply:
Linking to the task ab#69162